lib/mongoid/contextual/mongo.rb

```diff
@@ -376,12 +376,12 @@ def initialize(criteria)
       #
       # @return [ Document ] The last document.
       def last(limit_or_opts = nil)
-        limit = limit_or_opts unless limit_or_opts.is_a?(Hash)
+        limit, opts = extract_limit_and_opts(limit_or_opts)
         if cached? && cache_loaded?
           return limit ? documents.last(limit) : documents.last
         end
         res = try_numbered_cache(:last, limit) do
```

Comment:
I thought we got rid of this numbered cache in favor of the regular QueryCache? Do any of the other methods use it?

Reply:
This is 7.5-stable---confusing, I know! This is fixing a regression. It doesn't affect 8.